experiment/adscert/remotesigner.go

```diff
@@ -0,0 +1,54 @@
+package adscert
+
+import (
+	"fmt"
+	"github.com/IABTechLab/adscert/pkg/adscert/api"
+	"github.com/IABTechLab/adscert/pkg/adscert/signatory"
+	"github.com/prebid/prebid-server/config"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"time"
+)
+
+type remoteSigner struct {
+	signatory signatory.AuthenticatedConnectionsSignatory
+}
+
+func (rs *remoteSigner) Sign(destinationURL string, body []byte) (string, error) {
+	// The RequestInfo proto contains details about the individual ad request
+	// being signed.  A SetRequestInfo helper function derives a hash of the
+	// destination URL and body, setting these value on the RequestInfo message.
+	reqInfo := &api.RequestInfo{}
+	signatory.SetRequestInfo(reqInfo, destinationURL, []byte(body))
+
+	// Request the signature.
+	signatureResponse, err := rs.signatory.SignAuthenticatedConnection(
+		&api.AuthenticatedConnectionSignatureRequest{
+			RequestInfo: reqInfo,
+		})
+	if err != nil {
+		return "", err
+	}
+	if signatureResponse != nil && signatureResponse.SignatureOperationStatus == api.SignatureOperationStatus_SIGNATURE_OPERATION_STATUS_OK {
```

Comment:
Rather than checking for `signatureResponse != nil` explicitly here, you can just call the `GetSignatureOperationStatus()` which already accounts for that condition. Similar to what you're doing for `InProcessSigner`

Reply:
Definitely agree with this! Refactored